maven-core/src/main/java/org/apache/maven/lifecycle/internal/LifecycleModuleBuilder.java

```diff
@@ -89,8 +89,12 @@ public void buildProject( MavenSession session, MavenSession rootSession, Reacto
 
         // session may be different from rootSession seeded in DefaultMaven
         // explicitly seed the right session here to make sure it is used by Guice
-        sessionScope.enter( reactorContext.getSessionScopeMemento() );
```

Comment:
I have no idea what `getSessionScopeMemento()`, but can you explain why it is not required anymore?

Reply:
The memento was introduced in https://github.com/apache/maven/commit/45563ff5cbfd066761913ae35a4e3e8e6e25e83c as a fix for the bad behavior of the session scope with concurrent builds.  However, the fix is wrong, mainly because the scope implementation does not support multiple threads : it suppose a session is mapped to a single thread.  To fix the behavior for concurrent builds, the memento was added to carry over data from one thread to another, but this broke the scope.  This commit aims at fixing the scope implementation correctly and thus there's no need for the memento anymore.

Reply:
Very decent explanation.